kubernetes/resource_kubernetes_job.go

```diff
@@ -97,6 +111,11 @@ func resourceKubernetesJobUpdate(d *schema.ResourceData, meta interface{}) error
 	log.Printf("[INFO] Submitted updated job: %#v", out)
 
 	d.SetId(buildId(out.ObjectMeta))
+
+	if d.Get("wait_for_completion").(bool) {
+		return resource.Retry(d.Timeout(schema.TimeoutCreate),
```

Comment:
I think this should follow the value of `schema.TimeoutUpdate` instead. See my suggestion above about adding one to the schema.